packages/flutter_tools/lib/src/commands/daemon.dart

```diff
@@ -339,25 +339,30 @@ class AppDomain extends Domain {
       );
     }
 
+    bool supportsRestart = hotMode ? device.supportsHotMode : device.supportsRestart;
```

Comment:
That seems simplest - adding more info to the return value of app.restart. We could instead fire async events - like we do for the debug port value, or stdout messages. But that would probably only make sense if the state could change at any time.